docs/tools.md

```diff
@@ -0,0 +1,328 @@
+## OpenVoiceOS Agent Tools Documentation
+
+This documentation describes the **Agent Tools** framework in OpenVoiceOS (OVOS), a standardized mechanism for providing discoverable and executable functions to intelligent agents (e.g., LLM solvers, persona plugins).
+
+The primary goal is to provide a **generic discovery and execution interface** for tools, decoupling the tool logic from the consumption logic. Agents can discover tools and their contracts (input/output schemas) via the MessageBus or call them directly in Python.
+
+-----
+
+### Core Concepts
+
+#### 1. AgentTool
+
+The fundamental unit of the framework. An `AgentTool` is a simple data structure that defines a single, executable function and its contract.
+
+| Attribute | Type | Description |
+| :--- | :--- | :--- |
+| **`name`** | `str` | The **unique**, snake_case identifier for the tool. This name is typically used by the agent/LLM to reference the function. |
```

Comment:
Is `snake_case` strictly enforced or just suggested?

Reply:
just seems to be how everyone is doing it, but there's no specific checks to enforce it